internal/catchup_fetch_handler.go

```diff
@@ -0,0 +1,42 @@
+package internal
+
+import (
+	"fmt"
+	"wal-g/utility"
+
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/tracelog"
+)
+
+func catchupRecursion(backupName string, folder storage.Folder, dbDir string, fromLSN uint64) (uint64, error) {
+	backup, err := GetBackupByName(backupName, folder)
+	if err != nil {
+		return 0, err
+	}
+	sentinelDto, err := backup.GetSentinel()
+	if err != nil {
+		return 0, err
+	}
+
+	if sentinelDto.IsIncremental() && *sentinelDto.BackupStartLSN < fromLSN {
```

Comment:
You have 4 incremental backups:
001
002
003
004
If you want catchup from 002 it will check that 004 > 002, skip this and finish